pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -164,6 +166,9 @@ type EndpointController struct {
 	triggerTimeTracker *endpointutil.TriggerTimeTracker
 
 	endpointUpdatesBatchPeriod time.Duration
+
+	// serviceSelectorCache is a cache of service selector to avoid high cpu consume caused by AsSelectorPreValidated frequently called #73527
```

Comment:
i would reword to:
> serviceSelectorCache is a cache of service selectors to avoid high CPU consumption caused by frequent calls to AsSelectorPreValidated (see #73527)

[1]